evm_loader/performance/run.py

```diff
@@ -0,0 +1,589 @@
+
+def check_transfer_event(result, erc20_eth, acc_from, acc_to, sum, return_code):
+    assert(result['meta']['err'] == None)
+    assert(len(result['meta']['innerInstructions']) == 1)
+    assert(len(result['meta']['innerInstructions'][0]['instructions']) == 2)
+    data = b58decode(result['meta']['innerInstructions'][0]['instructions'][1]['data'])
+    assert(data[:1] == b'\x06')  #  OnReturn
+    assert(data[1:2] == return_code)    # 11 - Machine encountered an explict stop,
+                                        # 12 - Machine encountered an explict return
+    data = b58decode(result['meta']['innerInstructions'][0]['instructions'][0]['data'])
+    assert(data[:1] == b'\x07')  # 7 means OnEvent
+    assert(data[1:21] == bytes.fromhex(erc20_eth))
+    assert(data[21:29] == bytes().fromhex('%016x' % 3)[::-1])  # topics len
+    assert(data[29:61] == abi.event_signature_to_log_topic('Transfer(address,address,uint256)'))  # topics
+    assert(data[61:93] == bytes().fromhex("%024x" % 0) + bytes.fromhex(acc_from))  # from
+    assert(data[93:125] == bytes().fromhex("%024x" % 0) + bytes.fromhex(acc_to))  # to
+    assert(data[125:157] == bytes().fromhex("%064x" % sum))  # value
+    return True
+
+def get_filehash(factory, factory_code, factory_eth, acc):
+    trx = Transaction()
+    trx.add(
+        TransactionInstruction(
+            program_id=evm_loader_id,
+            data=bytearray.fromhex("03") + abi.function_signature_to_4byte_selector('get_hash()'),
+            keys=[
+                AccountMeta(pubkey=factory, is_signer=False, is_writable=True),
+                AccountMeta(pubkey=factory_code, is_signer=False, is_writable=True),
+                AccountMeta(pubkey=acc.public_key(), is_signer=True, is_writable=False),
+                AccountMeta(pubkey=evm_loader_id, is_signer=False, is_writable=False),
+                AccountMeta(pubkey=PublicKey(sysvarclock), is_signer=False, is_writable=False),
+            ]))
+    result = send_transaction(client, trx, acc)['result']
+
+    assert(result['meta']['err'] == None)
+    assert(len(result['meta']['innerInstructions']) == 1)
+    assert(len(result['meta']['innerInstructions'][0]['instructions']) == 2)
+    data = b58decode(result['meta']['innerInstructions'][0]['instructions'][1]['data'])
+    assert(data[:1] == b'\x06')  #  OnReturn
+    assert(data[1] == 0x11)  # 11 - Machine encountered an explict stop
+
+    data = b58decode(result['meta']['innerInstructions'][0]['instructions'][0]['data'])
+    assert(data[:1] == b'\x07')  # 7 means OnEvent
+    assert(data[1:21] == factory_eth)
+    assert(data[21:29] == bytes().fromhex('%016x' % 1)[::-1])  # topics len
+    hash = data[61:93]
+    return hash
+
+
+def get_trx(contract_eth, caller, caller_eth, input, pr_key):
+    tx = {'to': contract_eth, 'value': 1, 'gas': 1, 'gasPrice': 1,
+        'nonce': getTransactionCount(client, caller), 'data': input, 'chainId': chain_id}
```

Comment:
You must be able to generate multiple transactions for one account. Therefore, it is worth keeping the expected nonce and using it. If this information is not available, then ask the node.

Reply:
fixed